client/Client.cpp

```diff
@@ -593,7 +598,7 @@ void CClient::battleStarted(const BattleInfo * info)
 
 	if(info->tacticDistance && vstd::contains(battleints, info->sides[info->tacticsSide].color))
 	{
-		boost::thread(&CClient::commenceTacticPhaseForInt, this, battleints[info->sides[info->tacticsSide].color]);
+		playerTacticThreads[info->sides[info->tacticsSide].color] = std::make_unique<boost::thread>(boost::thread(&CClient::commenceTacticPhaseForInt, this, battleints[info->sides[info->tacticsSide].color]));

```

Comment:
Do we need to specify it once again here if we already said it in std::make_unique<boost::thread>? I am not c++ pro, just want to ensure there will be no copy-construction here

Reply:
std::make_unique<boost::thread>(boost::thread(&CClient::commenceTacticPhaseForInt, this, battleints[info->sides[info->tacticsSide].color]));
```suggestion
		playerTacticThreads[info->sides[info->tacticsSide].color] = std::make_unique<boost::thread>(&CClient::commenceTacticPhaseForInt, this, battleints[info->sides[info->tacticsSide].color]);

```

Reply:
I believe boost::thread is already non-copyable, so copy-construction should be impossible.
However this line is very overloaded. Perhaps split it?
```suggestion
		PlayerColor color = info->sides[info->tacticsSide].color;
		playerTacticThreads[color] = std::make_unique<boost::thread>(&CClient::commenceTacticPhaseForInt, this, battleints[color]);

```

Reply:
Yes this looks better